### PR TITLE
Feat: Section Component

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

New component for the new neuron setting page: PageSection

**PENDING: Design approval**

# Changes

* New component PageSection.
* Add new component to docs.

# Screenshots

Screenshot of the documentation page with the showcase:

section-3

Screenshot of the mobile version of the real production showcase

Screenshot 2023-07-11 at 07 52 36

### CODE REVIEW

**Comment by lmuntaner** [comment]
@mstrasinskis please review

File: src/lib/components/PageSection.svelte [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<script lang="ts">
+  export let title: string;
+  export let testId: string | undefined = undefined;
+</script>
+
+<div class="container" data-tid={testId}>
+  <div class="title-wrapper">
+    <h3 class="title">{title}</h3>
+    <slot name="value" />
```
Comment (mstrasinskis) [inline src/lib/components/PageSection.svelte:9]: Probably the slot `value` could have a more generic name. Or the component name could be more specific 🤔

File: src/lib/components/PageSection.svelte [hunk 2]
```diff
@@ -0,0 +1,35 @@
+<script lang="ts">
+  export let title: string;
+  export let testId: string | undefined = undefined;
+</script>
+
+<div class="container" data-tid={testId}>
+  <div class="title-wrapper">
+    <h3 class="title">{title}</h3>
```
Comment (mstrasinskis) [inline src/lib/components/PageSection.svelte:8]: Class `.title` is not in use.

Comment (mstrasinskis) [reply src/lib/components/PageSection.svelte:8]: Would it make sense to provide a slot to be more flexible (h2, h4)?

**Review by mstrasinskis — COMMENTED** [review]
Just a couple of comments. It feels a bit like some hierarchy information is missing. What is a page, Where can it be used. Which components are allowed to be on the page. Alternatively this could be a VerticalContainer with some slots...

File: src/lib/components/PageSection.svelte (hunk 1 above)
Comment (lmuntaner) [reply src/lib/components/PageSection.svelte]: Yes, I renamed it to `end` following the convention that @peterpeterparker mentioned

File: src/lib/components/PageSection.svelte (hunk 2 above)
Comment (lmuntaner) [reply src/lib/components/PageSection.svelte:8]: I wanted to discuss this with Artem. But it makes sense actually.

This way there can be sections of different levels maybe. I'll change it, thanks!

File: src/lib/components/PageSection.svelte (hunk 1 above)
Comment (mstrasinskis) [reply src/lib/components/PageSection.svelte]: Thanks 👍

**Comment by lmuntaner** [comment]
@mstrasinskis ready for another review.

I renamed it to Section

**Comment by artkorotkikh-dfinity** [comment]
@lmuntaner can we please adjust a few spacings in this component to something like this? (I also adjusted on Figma): 
image

Questions: 
1. Is it safe to assume that this component will support things like "copy address" in `<span slot="value" class="value">`? 
image

**Comment by lmuntaner** [comment]
@artkorotkikh-dfinity thanks for the review!

I did the changes.

Yes, the copy button can be added. That's another component unrelated to this one: `KeyValuePair`. It's already in this library and has worked bery well so far.

**Pull request merged**